AzureCalling/app/src/main/java/com/azure/samples/communication/calling/externals/authentication/AADAuthHandler.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.samples.communication.calling.externals.authentication;
+
+import static com.azure.samples.communication.calling.contracts.Constants.DISPLAY_NAME;
+import static com.azure.samples.communication.calling.contracts.Constants.GIVEN_NAME;
+import static com.azure.samples.communication.calling.contracts.Constants.ID;
+import static com.azure.samples.communication.calling.contracts.SampleErrorMessages.USER_LOGIN_CANCEL;
+
+import android.app.Activity;
+import android.os.Build;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.azure.samples.communication.calling.R;
+import com.azure.samples.communication.calling.utilities.AppSettings;
+import com.azure.samples.communication.calling.utilities.MSGraphRequestWrapper;
+import com.microsoft.identity.client.AcquireTokenSilentParameters;
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.AzureCloudInstance;
+import com.microsoft.identity.client.IAccount;
+import com.microsoft.identity.client.IAuthenticationResult;
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.ISingleAccountPublicClientApplication;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.SilentAuthenticationCallback;
+import com.microsoft.identity.client.exception.MsalException;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class AADAuthHandler {
+
+    private static final String LOG_TAG = AADAuthHandler.class.getSimpleName();
+
+    private final AppSettings appSettings;
+    private ISingleAccountPublicClientApplication mSingleAccountApp;
+    private String accessToken = null;
+    private String[] mScopes = { "User.Read" };
```

Comment:
If this is required for graph API then we can create new scope variable. As this scope is for AAD.

We can rename this class to AAD+Graph or can create separate classes for handling graph and AAD. Creating separate will be better. Please sync with iOS team.